backend/tournesol/models/entity_poll_rating.py

```diff
@@ -67,4 +75,38 @@ def update_n_ratings(self):
             .distinct("user")
             .count()
         )
+
         self.save(update_fields=["n_comparisons", "n_contributors"])
+
+    @staticmethod
+    def bulk_update_sum_trust_scores(poll: Poll, batch_size: Optional[int] = 4000):
+        if batch_size is None:
+            EntityPollRating._bulk_update_sum_trust_score(poll)
+        else:
+            EntityPollRating._bulk_update_sum_trust_score_by_batch(poll, batch_size)
+
+    @staticmethod
+    def _bulk_update_sum_trust_score(poll):
+        EntityPollRating.objects.filter(poll=poll).update(
+            sum_trust_scores=Coalesce(Subquery(
+                Entity.objects.filter(
+                    id=OuterRef('entity_id'),
+                    contributorvideoratings__poll=OuterRef('poll'),
+                ).annotate(
+                    s_t_s=Sum("contributorvideoratings__user__trust_score")
+                ).values("s_t_s")[:1]
+            ), 0.)
+        )
```

Comment:
This approach does not seem quite correct: for a given pair (entity, user) a `ContributorRating` may exist in the database even if the user has not evaluated the entity (for example when the user marked their rating as private or public, without submitting the comparison). 

So only ratings where at least one  `ContributorRatingCriteriaScore` exists should be considered (or almost equivalently, where a  `ContributorRatingCriteriaScore` on the main criterion exists);

This implementation should work:
```py
       EntityPollRating.objects.filter(poll=poll).update(
            sum_trust_scores=Coalesce(
                Subquery(
                    ContributorRating.objects.filter(
                        entity_id=OuterRef("entity_id"),
                        poll=OuterRef("poll"),
                    )
                    .filter(
                        Exists(
                            ContributorRatingCriteriaScore.objects.filter(
                                contributor_rating=OuterRef("id")
                            )
                        )
                    )
                    .values("entity")
                    .annotate(s_t_s=Sum("user__trust_score"))
                    .values("s_t_s")[:1]
                ),
                0.0,
            )
        )
 ```

Reply:
Addressed in e005d9a

Reply:
Thanks for the review and catching this issue @amatissart !
Merging now, and opening the next steps which will change the recommendations.